tests/Monolog/Formatter/JsonFormatterTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ErrorControl\Tests\Monolog\Formatter;
+
+use DateTimeImmutable;
+use Exception;
+use Keboola\ErrorControl\Monolog\Formatter\JsonFormatter;
+use Monolog\Level;
+use Monolog\LogRecord;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+
+class JsonFormatterTest extends TestCase
+{
+    public function testFormatting(): void
+    {
+        $logWithException = $this->givenLogWithException();
+        $formattedOutput = $this->whenLogIsFormatted($logWithException);
+        $this->thenOutputFormatIsCorrect($formattedOutput);
+    }
+
+    private function givenLogWithException(): LogRecord
+    {
+        return new LogRecord(
+            new DateTimeImmutable(),
+            'test',
+            Level::Critical,
+            'Message',
+            [new Exception('Message')]
+        );
+    }
+
+    private function whenLogIsFormatted(LogRecord $logRecord): string
+    {
+        return (new JsonFormatter())->format($logRecord);
+    }
+
+    private function thenOutputFormatIsCorrect(string $output): void
+    {
+        $json = json_decode($output);
+        assert($json instanceof stdClass);
```

Comment:
Mas nejaky duvod pro pouziti obycejneho `assert` misto `self::assertInstanceOf`?

Reply:
Jasny, chapu.. nech to tu. Spis jen tak pro zamysleni to budoucna, jak na to koukam:
* PHPStan to pochopi i z toho PHPUnitovyho assertu
* plain assert musi byt vhodne nakonfigurovany v `php.ini`
* ty vis, co si tam poslal, protoze si psal ten test.. ale kdyz k tomu za tyden prijde nekdo jiny, muze pres `$output` poslat neco jinyho (je to proste jen `string`)

Vlastne jedina "nevyhoda" toho pouzit PHPUnit assert je, ze takle (plain assert) mas rozliseny, co ocekavas "programatorsky" a co je ten test. Ale v praxi to nejak zasadni vliv IMHO nema. Podobne jsem to po nastupu taky zkousel :smile: ale casem jsem si zvykl v testech pouzivat vyhradne PHPunit asserty (tak jak to bylo v Keboole zajety).